IndicoToolkit/Extractions.cs

```diff
@@ -0,0 +1,166 @@
+using System.Linq;
+using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
+using Indico;
+using Indico.Jobs;
+using System.Threading.Tasks;
+
+using IndicoToolkit.Types;
+
+namespace IndicoToolkit
+{
+    /// <summary>
+    /// Class <c>Extractions</c> provides functionality for common extraction prediction use cases.
+    /// </summary>
+    public class Extractions {
+        internal List<Prediction> preds = new List<Prediction>();
+        public List<Prediction> removed_predictions = new List<Prediction>();
+        
+        public Extractions(List<Prediction> predictions)
+        {
+            this.preds = predictions;
+        }
+
+        public List<Prediction> Preds() 
+        {
+            return preds;
+        }
+
+        /// <summary>
+        /// Method <c>toDictByLabel</c> generates a dictionary where key is label string and value is list of all predictions of that label.
+        /// </summary>
+        public Dictionary<string, List<Prediction>> toDictByLabel()
+        {
+            Dictionary<string, List<Prediction>> prediction_label_map = new Dictionary<string, List<Prediction>>();
+            for (int i = 0; i < preds.Count; i++) 
+            {
+                Prediction pred = preds[i];
+                string pred_label = pred.getValue("label");
+
+                if (prediction_label_map.ContainsKey(pred_label)) {
+                    prediction_label_map[pred_label].Append(pred);
+                } else {
+                    prediction_label_map.Add(pred_label, new List<Prediction>() {pred});
+                }
+            }
+            return prediction_label_map;
+        }
+        
+        /// <summary>
+        /// Method <c>numPredictions</c> returns number of predictions.
+        /// </summary>
+        public int numPredictions() 
+        {
+            return preds.Count; 
+        }
+
+        /// <summary>
+        /// Method <c>removeByConfidence</c> removes predictions that are less than given confidence.
+        /// </summary>
+        public void removeByConfidence(float confidence = 0.95f, List<string> labels = null) 
+        {
+            List<Prediction> high_conf_preds = new List<Prediction>();
+            for (int i = 0; i < preds.Count; i++) {
+                Prediction pred = preds[i];
+                string pred_label = pred.getValue("label");
+                if (labels != null && !labels.Contains(pred_label)) {
+                    high_conf_preds.Append(pred);
+                } else if (pred.getValue("confidence")[pred_label] >= confidence) {
+                    high_conf_preds.Append(pred);
+                } else {
+                    removed_predictions.Append(pred);
```

Comment:
curious for your thoughts, but I find it much cleaner to read if this is `this.removed_predictions` so that it's immediately obvious that the variable is an attribute on the class

Reply:
Sounds good. I'm assuming we should do the same for preds then?

Reply:
yup